tests/test_states.py

```diff
@@ -220,6 +247,394 @@ def test_squeezed_coherent(self):
+
+        mu = np.zeros([6])
+        cov = np.zeros([6, 6])
+
+        # construct the vector of means and covariance matrix for each mode
+        # note that we use (x0,p0,x1,p1,...) ordering here.
+        for i, (x, y, z) in enumerate(zip(self.a_list, self.r_list, self.phi_list)):
+            mu[2*i:2*i+2] = R(self.qphi).T @ np.array([x.real, x.imag]) * np.sqrt(2*self.hbar)
+            cov[2*i:2*i+2, 2*i:2*i+2] = R(self.qphi).T @ squeezed_cov(y, z, hbar=self.hbar) @ R(self.qphi)
+
+        # apply a beamsplitter to vector of means and covariance matrices
+        t = 1/np.sqrt(2)
+        BS = np.array([[ t,  0,  -t,  0],
+                       [ 0,  t,  0,  -t],
+                       [ t,  0,  t,  0],
+                       [ 0,  t,  0,  t]])
+
+        S1 = block_diag(BS, np.identity(2))
+
+        if self.linear_modes == 3:
+            S2 = block_diag(np.identity(2), BS)
+        elif self.linear_modes == 2:
+            S2 = np.identity(6)
+
+        self.mu = S2 @ S1 @ mu
+        self.cov = S2 @ S1 @ cov @ S1.T @ S2.T
+
+    def test_multi_mode_linear_combination(self):
+        """Test that the correct result is returned for E(2x0 + x1 - p0)"""
+        self.logTestName()
+
+        x0 = 0.123
+        x1 = 2.2342
+        x2 = 1.
+        p0 = -5/np.sqrt(2)
+
+        # define the linear combination
+        A = np.zeros([6, 6])
+
+        if self.linear_modes == 3:
+            d = np.array([x0, x1, x2, p0, 0, 0])
+        elif self.linear_modes == 2:
+            d = np.array([x0, x1, 0, p0, 0, 0])
+
+        k = 0
+
+        self.circuit.reset(pure=self.kwargs['pure'])
+
+        # squeeze and displace each mode
+        for i, (x, y, z) in enumerate(zip(self.a_list, self.r_list, self.phi_list)):
+            self.circuit.prepare_displaced_squeezed_state(x, y, z, i)
+
+        # apply a beamsplitter to the modes
+        self.circuit.beamsplitter(1/np.sqrt(2), 1/np.sqrt(2), 0, 1)
+
+        if self.linear_modes == 3:
+            self.circuit.beamsplitter(1/np.sqrt(2), 1/np.sqrt(2), 1, 2)
+
+        state = self.circuit.state()
+        mean, var = state.poly_quad_expectation(A, d, k, phi=self.qphi)
+
+        # E(2x0 + x1 - p0) = 2E(x0) + E(x1) - E(p0)
+        mean_expected = np.array([x0, p0, x1, 0, x2, 0]) @ self.mu
+        self.assertAlmostEqual(mean, mean_expected, delta=self.tol)
+
+        # var(x0 + x1 + p0) = 4var(x0)+var(x1)+var(p0)-4cov(x0,p0) + 4cov(x0, x1) - 2*cov(x1, p0)
+        var_expected = self.cov[0, 0]*x0**2 + self.cov[1, 1]*p0**2 + self.cov[2, 2]*x1**2 \
+            + self.cov[4, 4]*x2**2 + 2*self.cov[0, 1]*x0*p0 + 2*self.cov[0, 2]*x0*x1 \
+            + 2*self.cov[0, 4]*x0*x2 + 2*self.cov[1, 2]*p0*x1 + 2*self.cov[1, 4]*p0*x2 \
+            + 2*self.cov[2, 4]*x1*x2
+        self.assertAlmostEqual(var, var_expected, delta=self.tol)
+
+    def test_multi_mode_quadratic_combination(self):
+        """Test that the correct result is returned for E(x0x1+x0p1)"""
```

Comment:
See earlier comment about testing only symmetric operators

Reply:
Same as above; in this case, E(x0x1+x0p1) is the polynomial in the phase space, not the quantized operators